packages/mds-ingest-service/@types/models.ts

```diff
@@ -0,0 +1,148 @@
+import type { DomainModelCreate, IdentityColumn, RecordedColumn } from '@mds-core/mds-repository'
+import type {
+  ACCESSIBILITY_OPTION,
+  MODALITY,
+  Nullable,
+  PROPULSION_TYPE,
+  TelemetryData,
+  Timestamp,
+  TRIP_STATE,
+  UUID,
+  VEHICLE_EVENT,
+  VEHICLE_STATE,
+  VEHICLE_TYPE
+} from '@mds-core/mds-types'
+export interface DeviceDomainModel extends RecordedColumn {
+  device_id: UUID
+  provider_id: UUID
+  vehicle_id: string
+  vehicle_type: VEHICLE_TYPE
+  propulsion_types: PROPULSION_TYPE[]
+
+  year: Nullable<number>
+  mfgr: Nullable<string>
+  model: Nullable<string>
+  accessibility_options: Nullable<ACCESSIBILITY_OPTION[]>
+  modality: MODALITY
+}
+
+export type DeviceDomainCreateModel = DomainModelCreate<Omit<DeviceDomainModel, keyof RecordedColumn>>
+
+export type GpsData = Omit<TelemetryData, 'charge'>
+
+export interface EventDomainModel extends RecordedColumn {
+  device_id: UUID
+  provider_id: UUID
+  timestamp: Timestamp
+  event_types: VEHICLE_EVENT[]
+  vehicle_state: VEHICLE_STATE
+  trip_state: Nullable<TRIP_STATE>
+
+  telemetry_timestamp: Timestamp
+  telemetry: TelemetryDomainModel
+  annotation: Nullable<EventAnnotationDomainModel>
+  trip_id: Nullable<UUID>
+}
+
+export type EventDomainCreateModel = DomainModelCreate<
+  Omit<EventDomainModel, keyof RecordedColumn | 'telemetry' | 'telemetry_timestamp'>
+> & {
+  telemetry: TelemetryDomainCreateModel
+}
+
+export type MigratedEventDomainModel = Omit<EventDomainModel & IdentityColumn, 'telemetry' | 'annotation'>
+
+export interface TelemetryDomainModel extends RecordedColumn {
+  device_id: UUID
+  provider_id: UUID
+  timestamp: Timestamp
+  gps: GpsData
+  charge: Nullable<number>
+  stop_id: Nullable<UUID>
+}
+
+export type TelemetryDomainCreateModel = DomainModelCreate<Omit<TelemetryDomainModel, keyof RecordedColumn | 'gps'>> & {
+  gps: DomainModelCreate<GpsData>
+}
+
+export const GROUPING_TYPES = ['latest_per_vehicle', 'latest_per_trip', 'all_events'] as const
+export type GROUPING_TYPE = typeof GROUPING_TYPES[number]
```

Comment:
Some of the things that were moved to this new `models.ts` file seem more like query options, might make sense to put them in a separate file perhaps